File: base/python/hashtable.py
import os
import time
from random import seed
from random import randint

import uuid

# seed random number generator
seed(time.time())

def compute():

    size = int(os.environ['HASHTABLE_SIZE'])
    read_count = int(os.environ['READ_COUNT'])

    # create an input map and an array
    hashtable = {}
    keys = [0] * size

    # init the map
    for i in range(size):
        key = uuid.uuid1()
        hashtable[key]=key
        keys[i]=key

    #access map
    for i in range(read_count):
        key = keys[randint(0, size-1)]
        _ = hashtable[key]

File: base/go/hashtable.go
package main

import (
	"math/rand"
	"os"
	"strconv"

	"github.com/google/uuid"
)

func compute() {

	size, _ := strconv.Atoi(os.Getenv("HASHTABLE_SIZE"))
	readCount, _ := strconv.Atoi(os.Getenv("READ_COUNT"))

	//create an input hash table and an array
	hashtable := make(map[uuid.UUID]uuid.UUID, size)
	keys := make([]uuid.UUID, size)

	// init the hash table
	for i := 0; i < size; i++ {
		key := uuid.New()
		hashtable[key] = key
		keys[i] = key
	}

	//access hash table
	for i := 0; i < readCount; i++ {
		key := keys[rand.Intn(size)]
		_ = hashtable[key]
	}

}
